google/cloud/internal/win32/parse_service_account_p12_file.cc

```diff
@@ -142,23 +131,29 @@ std::string GetCertificateCommonName(PCCERT_CONTEXT cert) {
   return result;
 }
 
-StatusOr<UniqueCryptProv> GetCertificatePrivateKey(PCCERT_CONTEXT cert,
-                                                   DWORD& dwKeySpec,
-                                                   std::string const& source) {
-  HCRYPTPROV prov_raw;
-  BOOL pfCallerFreeProvOrNCryptKey;
-  if (!CryptAcquireCertificatePrivateKey(cert, CRYPT_ACQUIRE_SILENT_FLAG,
-                                         nullptr, &prov_raw, &dwKeySpec,
-                                         &pfCallerFreeProvOrNCryptKey)) {
+StatusOr<HCRYPTPROV> GetCertificatePrivateKey(PCCERT_CONTEXT cert,
+                                              DWORD& dwKeySpec,
+                                              std::string const& source) {
+  DWORD context_length;
+  if (!CertGetCertificateContextProperty(cert, CERT_KEY_CONTEXT_PROP_ID,
+                                         nullptr, &context_length)) {
     return InvalidArgumentError(
         FormatWin32Errors("No private key found in PKCS#12 file (", source,
                           "): "),
         GCP_ERROR_INFO());
   }
-  // According to documentation of CryptAcquireCertificatePrivateKey,
-  // pfCallerFreeProvOrNCryptKey will always be true in our case so
-  // we don't need to check it.
-  return UniqueCryptProv(prov_raw);
+  std::vector<BYTE> buffer(context_length);
+  CertGetCertificateContextProperty(cert, CERT_KEY_CONTEXT_PROP_ID,
```

Comment:
should we handle the case where this call fails?

It looks like the examples do: https://learn.microsoft.com/en-us/windows/win32/seccrypto/example-c-program-getting-and-setting-certificate-properties

Reply:
We check for errors in the first invocation of `CertGetCertificateContextProperty` to get the buffer's length. We could check again but I don't think it's strictly necessary. See also comment in line 182.

Reply:
Ack, thanks.